frontend/src/api/actions/users.js

```diff
@@ -56,8 +57,13 @@ export const setActiveUserRole = validatedApiDispatcher({
     name: "setActiveUserRole",
     description: "Sets the role of the active user",
     onErrorDispatch: e => deleteError(e.toString()),
-    dispatcher: payload => async dispatch => {
-        dispatch(setActiveUserRoleSuccess(payload));
+    dispatcher: (payload, options = {}) => async dispatch => {
+        await dispatch(setActiveUserRoleSuccess(payload));
+        if (!options.skipInit) {
```

Comment:
Same thing here, if the activeUserRole gets changed, all the following stage should be performed right?